### PR TITLE
Generate the client_config.json file

### PR DESCRIPTION
**Pull request by chrisdunelm**
This contains the retry information, derived from the grpc-service-config file

### CODE REVIEW

**Comment by chrisdunelm** [comment]
This isn't as large a PR as it looks - many of the files are generated by protoc, so can be ignored.

File: src/Generation/ResourcesGenerator.php [hunk 1]
```diff
@@ -66,4 +69,71 @@ public static function generateDescriptorConfig(ServiceDetails $serviceDetails):
...
+        $durationToMillis = fn($d) => (int)($d->getSeconds() * 1000 + $d->getNanos() / 1e6);
+
+        $retryCodes = $grpcServiceConfig->retryPolicies
+            ->map(fn($x, $i) => ["retry_policy_{$inc($i)}_codes", is_null($x) ? null :
+                Vector::new($x->getRetryableStatusCodes())->map(fn($x) => Code::name($x))->toArray()])
+            ->filter(fn($x) => !is_null($x[1]))
+            ->append(['no_retry_codes', []])
+            ->toArray(fn($x) => $x[0], fn($x) => $x[1]);
+
+            $retryParams = Vector::zip($grpcServiceConfig->retryPolicies, $grpcServiceConfig->timeouts, fn($r, $t, $i) =>
```
Comment (jskeet) [inline src/Generation/ResourcesGenerator.php:85]: This section looks like it's indented too far.

**Review by jskeet — APPROVED** [review]
Can't say I've looked in huge amounts of detail, but looks generally fine - one indentation nit.

File: src/Generation/ResourcesGenerator.php (hunk 1 above)
Comment (chrisdunelm) [reply src/Generation/ResourcesGenerator.php]: Thanks. Fixed.

**Pull request merged**